installation-manager/src/main/java/org/wildfly/core/instmgr/InstMgrCleanHandler.java

```diff
@@ -0,0 +1,108 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.core.instmgr;
+
+import java.nio.file.Path;
+import java.util.Collection;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationDefinition;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.installationmanager.Channel;
+import org.wildfly.installationmanager.MavenOptions;
+import org.wildfly.installationmanager.spi.InstallationManager;
+import org.wildfly.installationmanager.spi.InstallationManagerFactory;
+
+/**
+ * Operation handler that cleans the installation manager work directories or remove a custom patch.
+ */
+public class InstMgrCleanHandler extends InstMgrOperationStepHandler {
+    static final String OPERATION_NAME = "clean";
+    protected static final AttributeDefinition LIST_UPDATES_WORK_DIR = SimpleAttributeDefinitionBuilder.create(InstMgrConstants.LIST_UPDATES_WORK_DIR, ModelType.STRING)
+            .setRequired(false)
+            .setAlternatives(InstMgrConstants.CUSTOM_PATCH)
+            .setStorageRuntime()
+            .build();
+
+    protected static final AttributeDefinition CUSTOM_PATCH = SimpleAttributeDefinitionBuilder.create(InstMgrConstants.CUSTOM_PATCH, ModelType.BOOLEAN)
+            .setRequired(false)
+            .setAlternatives(InstMgrConstants.LIST_UPDATES_WORK_DIR)
+            .setStorageRuntime()
+            .build();
+
+    public static final OperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder(OPERATION_NAME, InstMgrResolver.RESOLVER)
+            .withFlags(OperationEntry.Flag.HOST_CONTROLLER_ONLY)
+            .addParameter(LIST_UPDATES_WORK_DIR)
+            .setRuntimeOnly()
+            .build();
+
+    InstMgrCleanHandler(InstMgrService imService, InstallationManagerFactory imf) {
+        super(imService, imf);
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        final String listUpdatesWorkDir = LIST_UPDATES_WORK_DIR.resolveModelAttribute(context, operation).asStringOrNull();
+        final boolean customPatch = CUSTOM_PATCH.resolveModelAttribute(context, operation).asBoolean(false);
```

Comment:
This comment/fix is now obsolete, we have changed the `clean` operation to specify an argument `custom-patch-manifest` instead of `custom-patch`. The new argument is an String instead of a boolean an specifies the Manifest GA of the custom-patch to remove